eng/scripts/Language-Settings.ps1

```diff
@@ -348,7 +369,47 @@ function Get-DocsCiLine ($item) {
   return $line
 }
 
+function EnsureCustomSource($package) {
+  # $PackageSourceOverride is a global variable provided in 
+  # Update-DocsMsPackages.ps1. Its value can set a "customSource" property.
+  # If it is empty then the property is not overridden.
+  $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+  if (!$customPackageSource) {
+    return $package
+  }
+
+  Write-Host "Checking custom package source for $($package.Name)"
+  $existingVersions = Find-Package `
+    -Name $package.Name `
+    -Source CustomPackageSource `
+    -AllVersions `
+    -AllowPrereleaseVersions
+  
+  # Matches package version against output: 
+  # "Azure.Security.KeyVault.Secrets 4.3.0-alpha.20210915.3"
+  $matchedVersion = $existingVersions.Where({$_.Version -eq $package.Versions})
+
+  if (!$matchedVersion) { 
+    return $package
+  }
+
+  $package.Properties['customSource'] = $customPackageSource
+  return $package
+}
+
 function Update-dotnet-DocsMsPackages($DocsRepoLocation, $DocsMetadata) {
+  # $PackageSourceOverride is a global variable provided in 
+  # Update-DocsMsPackages.ps1.
+  $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+  if ($customPackageSource) {
+    Write-Host "Registering custom package source $customPackageSource"
+    Register-PackageSource `
```

Comment:
Does this fail if the source is already registered?

Reply:
Using `-Force` overrides the failure. 

```pwsh
PS /home/djurek/azure-sdk-for-net> Register-PackageSource -Name CustomPackageSource -Provider NuGet -Location https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json

Register-PackageSource: Package Source 'CustomPackageSource' exists.
PS /home/djurek/azure-sdk-for-net> Register-PackageSource -Name CustomPackageSource -Provider NuGet -Location https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json -Force


Name                             ProviderName     IsTrusted  Location
----                             ------------     ---------  --------
CustomPackageSource              NuGet            False      https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json
```